### ISSUE TITLE
using pdf_documents for a single main doc results in two pdf outputs when using sphinx-build

### ISSUE THREAD

**Issue by droppedbars** [rst2pdf/rst2pdf#1213]
**Description of problem**
Using the sphinx integration, if you intend to only create a single pdf document and are using the pdf_documents object in conf.py, you will get two pdfs output.

To reproduce, setup a sphinx document, install and configure rst2pdf.pdfbuilder in it.
Update the conf.py with something similar to the following:

`project = 'project_name'`
`project_doc = 'doc_title'`
`copyright = '2024'`
`author = 'Author Name'`

`pdf_documents = [('index', 'pdf_name', 'doc_name', 'Author Name.'),]`

then run `make pdf`

You will get two pdfs:
* named 'pdf_name.pdf' with the title 'doc_name'
* named 'project_name.pdf' with the title 'project_name Documentation'

Also note, if you do not define the project and project_doc variables above, you'll get a funny result where the document is named 'Python'

In my local copy, I resolved this by updating `pdfbuilder.py` around lines 997 to 1006 and wrapped it in `if not app.config.pdf_documents:`

Essentially, if no `pdf_documents` object is not provided, or it has zero length, then it'll do the default behaviour of defining the pdf based off the conf.py project information. But, if a `pdf_documents` object is provided with a length greater than zero, then it skips that and uses only the contents of `pdf_documents` to generate the pdf(s).

My expectation is that if I'm configuring the document parameters via `pdf_documents` is that this would be the only source of the pdfs generated.

🖥 **Versions**

`python -V`
Python 3.10.12

`pip freeze | grep rst2pdf`
rst2pdf==0.101

`pip freeze | grep reportlab`
reportlab==4.2.0

`pip freeze | grep Sphinx`
Sphinx==7.3.7

Which operating system are you using?
Ubuntu 22.04

**Additional information**
it's possible I'm missing the intention of this block, but it seems unnecessary to build duplicate pdfs or a generic uncontrolled one with a default naming convention when the user is specifically using pdf_documents to define their pdf documents.

**Comment by Rubeanster** [comment]
I've also came across this issue recently after bumping my Sphinx version. Testing it with Sphinx <7.3.0 seems to work as expected with only one pdf generated based on the `pdf_documents` arguments, but using Shinx >=7.3.0 will generate 2 pdfs. I also tried changing rst2pdf versions and that had no affect.

Removing the `pdf_documents` from my conf.py only generates one pdf for me as well. Is this the new expected behavior or maybe an issue with Sphinx? Should the usage documentation get updated if this is expected?
